docs/milestone-1.md

```diff
@@ -0,0 +1,175 @@
+- error: code and message set in case an exception happens while proposing the payload.
+
+Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload), but we return the full payload for simplicity.
+
+### relay_getPayloadHeaderV1
+
+#### Request
+
+- method: `relay_getPayloadHeaderV1`
+- params:
+  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+
+#### Response
+
+- result: [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+- error: code and message set in case an exception happens while getting the payload.
+
+### relay_proposeBlindedBlockV1
+
+#### Request
+
+- method: `relay_proposeBlindedBlockV1`
+- params:
+  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
+  2. [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+
+#### Response
+
+- result: [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload)
+- error: code and message set in case an exception happens while proposing the payload.
+
+Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload), but we return the full payload for simplicity. 
+
+### Types
+
+#### SignedMEVPayloadHeader
+
+See [here](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/phase0/beacon-chain.md#custom-types) for the definition of fields like `BLSSignature`
+
+- `message`: [`MEVPayloadHeader`](#mevpayloadheader)
+- `signature`: `BLSSignature`
+
+#### MEVPayloadHeader
+
+- `payloadHeader`: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader)
+- `feeRecipient`: `Data`, 20 Bytes - the fee recipient address requested by the validator
+- `feeRecipientBalance`: `Quantity`, 256 Bits - the ending balance of the feeRecipient address
+
+Note: the feeRecipient must match the suggestedFeeRecipient address provided in the [`PayloadAttributesV1`](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#payloadattributesv1) of the associated [`engine_forkchoiceUpdatedV1`](#engine_forkchoiceupdatedv1) request.
+
+#### SignedBlindedBeaconBlock
+
+See [here](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/phase0/beacon-chain.md#custom-types) for the definition of fields like `BLSSignature`
+
+- `message`: [`BlindedBeaconBlock`](#blindedbeaconblock)
+- `signature`: `BLSSignature`
+
+#### BlindedBeaconBlock
+
+This is forked from [here](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/phase0/beacon-chain.md#beaconblock) with `body` replaced with `BlindedBeaconBlockBody`
```

Comment:
These references to the now-Bellatrix hardfork of the CL specs don't account for, e.g., supporting multiple CL MEV-compatible CL hardforks. While I'm not sure what the role of MEV in phase 0 or Altair might be, lacking transactions and all, the next hard fork, called the sharding fork pending whatever it's renamed to, will also support transactions.

The algorithm to derive the blinded types is the same, granted -- replace the execution payload with the execution payload header in the body, then propagate that back up the dependency chain -- but this creates ambiguous naming within a single global namespace as specified. In the context of MEV boosting, is a `BlindedBeaconBlock` (signed, body, etc) the Bellatrix or Sharding version?

Of course, that won't be for a while, but it's an eventuality that will happen and it'll be smoother to anticipate it now than later.

Reply:
AFAIK fork awareness hasn't been implemented in the engine API, and since this should be an extension of that repo I don't think it should be defined here. We should just roll with the precedent set upstream, which I imagine will be similar to what is used by the beacon API.